pkg/kapp/resources/resources.go

```diff
@@ -49,10 +59,10 @@ type Resources struct {
 	logger logger.Logger
 }
 
-func NewResources(resourceTypes ResourceTypes, coreClient kubernetes.Interface,
-	dynamicClient dynamic.Interface, fallbackAllowedNamespaces []string, logger logger.Logger) *Resources {
+func NewImpl(resourceTypes ResourceTypes, coreClient kubernetes.Interface,
+	dynamicClient dynamic.Interface, fallbackAllowedNamespaces []string, logger logger.Logger) *Impl {
```

Comment:
```suggestion
	dynamicClient dynamic.Interface, fallbackAllowedNamespaces []string, logger logger.Logger) Resources {
```

Reply:
current implementation is good. there is common saying: "Accept Interfaces and Return Concrete Types".

Reply:
Yep your right